core/arch/arm/include/mm/core_mmu_arch.h

```diff
@@ -210,6 +210,12 @@ static inline unsigned int core_mmu_get_va_width(void)
 	}
 	return 32;
 }
+
+static inline bool core_mmu_table_level_not_exceed(unsigned int level)
```

Comment:
How about making this:
``` C
static inline bool core_mmu_table_level_in_range(unsigned int level)
{
        return level >= CORE_MMU_BASE_TABLE_LEVEL &&
               level <= CORE_MMU_PGDIR_LEVEL;
}
```
With the advantage that we avoid confusion about which side we're checking in the interval.

Reply:
It makes sense.
But if `CORE_MMU_BASE_TABLE_LEVEL` equals 0, compiler will generate the warning:
```
warning: comparison of unsigned expression in ‘>= 0’ is always true [-Wtype-limits]
```
What about this version:
```
static inline bool core_mmu_table_level_in_range(unsigned int level)
{
#if CORE_MMU_BASE_TABLE_LEVEL == 0
	return level <= CORE_MMU_PGDIR_LEVEL;
#else
	return level >= CORE_MMU_BASE_TABLE_LEVEL &&
	       level <= CORE_MMU_PGDIR_LEVEL;
#endif
}
```

Reply:
OK.

Reply:
Resolved in new commit.